lib/etas/overlap_table_update.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require "services"
+require "utils/waypoint"
+require "date"
+require "etas/cluster_update"
+
+module ETAS
+
+  # An update of the table used for ETAS: holdings_htitem_htmember
+  class OverlapTableUpdate
+    attr_accessor :cutoff_date, :waypoint, :num_deletes, :num_adds
+
+    def initialize(cutoff_date = nil)
+      @cutoff_date = cutoff_date || Date.today - 1.5
+      @waypoint = Utils::Waypoint.new(BATCH_SIZE)
+      @num_deletes = 0
+      @num_adds = 0
+    end
+
+    def keep_alive(session, seconds = 120)
+      Thread.new do
+        loop do
+          sleep(seconds)
+          session.client.command(refreshSessions: [session.session_id])
+        end
+      end
+    end
+
+    def overlap_table
+      Services.holdings_db[TABLENAME]
+    end
+
+    def run
+      cutoff_str = cutoff_date.strftime("%Y-%m-%d %H:%M:%S")
+      Services.logger.info "Upserting clusters last_modified after #{cutoff_str} to #{TABLENAME}"
+      Cluster.with_session do |session|
+        session_refresh = keep_alive(session)
+        Cluster.where("ht_items.0": { "$exists": 1 },
+                      last_modified: { "$gt": cutoff_date }).no_timeout.each do |cluster|
+                        cu = ETAS::ClusterUpdate.new(overlap_table, cluster)
```

Comment:
Consider extracting this block to a separate method to better expose what it is we're doing here and appease rubocop

Reply:
Done